### DISCUSSION TITLE
Inbound Call Not Routing to PBX / No Response to SIP OPTIONS

### DISCUSSION THREAD

**Discussion by ph0nebooth** [hnimminh/libresbc#155] [Q&A] [answered]
Hi all

I've found some time to start experimenting with LibreSBC again and I've hit a roadblock. I'm attempting to sit LibreSBC in front of a FusionPBX instance. The issue I'm experiencing is that LibreSBC doesn't process incoming calls, despite receiving the SIP INVITE and it also doesn't respond to SIP OPTIONS from my PBX.

SBC Network
eth0 - 175.x.x.x (WAN, no NAT)
eth1 - 10.x.x.1 (LAN)

PBX Network
eth0 - 10.x.x.2 (LAN. WAN via 1:1 NAT)

Inbound call flow: Carrier -> LibreSBC eth0 -> LibreSBC eth1 -> PBX eth0
Outbound call flow: UAC -> PBX eth0 -> LibreSBC eth1 -> LibreSBC eth0 -> Carrier

The config has been reversed for the outbound leg from the PBX to Libre and beyond but I haven't provided that. I'm putting a focus on the inbound calls.

- I have configured both eth0 and eth1 as netalias within LibreSBC.
- I have a PBX and Carrier media class.
- I have an internal SIP Profile (below)
> {
    "name": "internal",
    "desc": "SIP Private Profile",
    "user_agent": "SBC",
    "sdp_user": "SBC",
    "local_network_acl": "rfc1918.auto",
    "addrdetect": "autonat",
    "enable_100rel": true,
    "ignore_183nosdp": true,
    "sip_options_respond_503_on_busy": false,
    "disable_transfer": true,
    "manual_redirect": true,
    "enable_3pcc": false,
    "enable_compact_headers": false,
    "dtmf_type": "rfc2833",
    "media_timeout": 0,
    "rtp_rewrite_timestamps": true,
    "context": "carrier",
    "sip_port": 5060,
    "sip_address": "eth1",
    "rtp_address": "eth1",
    "tls": false,
    "tls_only": false,
    "sips_port": 5061,
    "tls_version": "tlsv1.2",
    "realm": "internal.libresbc",
    "engagements": [
        "in:InterconSwitchWAInbound",
        "out:InterconSwitchWAOut"
    ]
}
- External SIP profile (below)
> {
    "name": "external",
    "desc": "SIP Public Profile",
    "user_agent": "SBC",
    "sdp_user": "SBC ",
    "local_network_acl": "rfc1918.auto",
    "addrdetect": "autonat",
    "enable_100rel": true,
    "ignore_183nosdp": true,
    "sip_options_respond_503_on_busy": false,
    "disable_transfer": true,
    "manual_redirect": true,
    "enable_3pcc": false,
    "enable_compact_headers": false,
    "dtmf_type": "rfc2833",
    "media_timeout": 0,
    "rtp_rewrite_timestamps": true,
    "context": "carrier",
    "sip_port": 5060,
    "sip_address": "eth0",
    "rtp_address": "eth0",
    "tls": false,
    "tls_only": false,
    "sips_port": 5061,
    "tls_version": "tlsv1.2",
    "realm": "external.libresbc",
    "engagements": [
        "out:InterconCrazyTelOut",
        "in:InterconCrazyTelInbound"
    ]
}
- I have an inbound interconnection for the carrier.
> {
    "translation_classes": [],
    "enable": true,
    "ringready": false,
    "rtpaddrs": [
        "43.245.163.222/32"
    ],
    "manipulation_classes": [],
    "nodes": [
        "_ALL_"
    ],
    "authscheme": "IP",
    "media_class": "pcm",
    "sipaddrs": [
        "43.245.163.222/32"
    ],
    "name": "InterconCrazyTelInbound",
    "capacity_class": "small",
    "sipprofile": "external",
    "routing": "topbx",
    "desc": "CrazyTel"
}
- I have an outbound interconnection for the PBX
> {
    "name": "InterconSwitchWAOut",
    "desc": "WA Call Switch",
    "sipprofile": "internal",
    "distribution": "weight_based",
    "sipaddrs": [
        "10.x.x.2/32"
    ],
    "rtpaddrs": [
        "10.x.x.2/32"
    ],
    "media_class": "pbx",
    "capacity_class": "small",
    "translation_classes": [],
    "manipulation_classes": [],
    "privacy": [
        "none"
    ],
    "cid_type": "none",
    "nofailover_sip_codes": [],
    "nodes": [
        "_ALL_"
    ],
    "enable": true,
    "gateways": [
        {
            "name": "GWSwitchWA",
            "weight": 1
        }
    ],
    "engagements": [
        "table:topbx"
    ]
}
- I have a SIP Gateway for the PBX
> {
    "name": "GWSwitchWA",
    "desc": "WA Call Switch",
    "username": "none",
    "password": "none",
    "proxy": "10.x.x.2",
    "port": 5080,
    "transport": "udp",
    "do_register": false,
    "caller_id_in_from": true,
    "cid_type": "none",
    "ping": 5,
    "engagements": [
        "InterconSwitchWAOut"
    ]
}
- I have a SIP Gateway for the carrier.
> {
    "name": "GWCrazyTel",
    "desc": "",
    "username": "dummyuser",
    "password": "dummypass",
    "proxy": "sip.biz.carrier.net.au",
    "outbound_proxy": "perth.sip.carrier.net.au",
    "port": 5060,
    "transport": "udp",
    "do_register": false,
    "register_proxy": "sip.biz.carrier.net.au",
    "register_transport": "udp",
    "expire_seconds": 60,
    "retry_seconds": 30,
    "caller_id_in_from": true,
    "extension_in_contact": true,
    "ping": 5,
    "ping_max": 1,
    "ping_min": 1,
    "engagements": [
        "InterconCrazyTelOut"
    ]
}
- I have a routing table for incoming calls
> {
    "name": "topbx",
    "desc": "route call to PBX",
    "action": "route",
    "routes": {
        "primary": "InterconSwitchWAOut",
        "secondary": "InterconSwitchWAOut",
        "load": 100
    },
    "engagements": [
        "in:InterconCrazyTelInbound"
    ]
}
- I have an outgoing route table too, which is pretty much the same as the incoming except for the routes being different.

It doesn't look like anything is bound to port 5060 on either interface?

> root@sbcwa:/var/log# netstat -tuln
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State      
tcp        0      0 127.0.0.1:6379          0.0.0.0:*               LISTEN     
tcp        0      0 127.0.0.1:8080          0.0.0.0:*               LISTEN     
tcp        0      0 127.0.0.1:8021          0.0.0.0:*               LISTEN     
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN     
tcp        0      0 0.0.0.0:8443            0.0.0.0:*               LISTEN     
tcp6       0      0 ::1:6379                :::*                    LISTEN     
tcp6       0      0 :::4369                 :::*                    LISTEN     
tcp6       0      0 :::22                   :::*                    LISTEN     
tcp6       0      0 :::8088                 :::*                    LISTEN     
udp        0      0 0.0.0.0:68              0.0.0.0:*                          
udp        0      0 0.0.0.0:68              0.0.0.0:*                          
udp        0      0 127.0.0.1:323           0.0.0.0:*                          
udp6       0      0 ::1:323                 :::*                               
root@sbcwa:/var/log#

**Comment by ph0nebooth** [comment, marked as answer]
So this is a really odd resolution to the issue but here's what the problem was... The cluster CPS and concurrent call amount was set higher than the capacity class. Setting them to the same amount seems to have brought freeswitch/kamailio online.

On another note, where are the Kamailo log files? Because there was nothing useful displayed in any of the /var/log files.

How are the tone scripts meant to be written out for the Pre Answer class? I've tried 400/400,-200,400/400,-2000 with no success.

**Reply by hnimminh** [reply]
@ph0nebooth try this `%(2000,4000,440.0,480.0)`

https://developer.signalwire.com/freeswitch/FreeSWITCH-Explained/Modules/mod-dptools/gentones/TGML_6586573#lua

**Reply by ph0nebooth** [reply]
@hnimminh I managed to get it working, thank you anyway! Do you have any plans to introduce the ability to proxy SIP REGISTER messages to allow a switch/PBX to act as the directory?
